source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,315 @@
+.. _csharp-class-mapping:
+
+=============
+Class Mapping
+=============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+
+Overview
+--------
+
+In this guide, you can learn how to customize the way the {+driver-long+}
+maps BSON documents to and from {+language+} classes.
+
+Creating A Class Map
+--------------------
+
+When you use a class to represent data in a MongoDB collection, the
+{+driver-short+} automatically creates a class map that serializes or deserializes
+that data. It does this by matching the name of the field in the document to the
+name of the property in the class. You can choose to manually control the
```

Comment:
I believe it's just the name. Waiting on eng to confirm

Reply:
Got confirmation from engineering. It only serializes based on the name but the type does matter due to the way the serializers worked. I restructured a bit and added an Automatic Class Mapping section so that I could detail this a little more.